roxie/ccd/ccdserver.cpp

```diff
@@ -4557,8 +4557,9 @@ class CRemoteResultAdaptor : implements IEngineRowStream, implements IFinalRoxie
             //       But this could still cause too many reply packets on the fastlane
             //       (higher priority output Q), which may cause the activities on the 
             //       low priority output Q to not get service on time.
+            unsigned pmask = p->queryHeader().activityId & ROXIE_PRIORITY_MASK;
             if ((colocalArg == 0) &&     // not a child query activity??
-                    (p->queryHeader().activityId & (ROXIE_SLA_PRIORITY | ROXIE_HIGH_PRIORITY)) && 
+                    (pmask && (pmask != ROXIE_PRIORITY_MASK)) &&
```

Comment:
You only want to request fastlane for SLA/HIGH. This code might do that, but it took me several reads of it to be sure. Can it be coded more clearly?

Reply:
ok.
pmask != 0 => its SLA, HIGH or BG
pmask != ROXIE_PRIORITY_MASK => its not BG